### PR TITLE
Added nameof syntax in scaffolding data annotations attributes

### LINKED ISSUE #11744: RevEng: Use nameof in data annotations
Currently when creating an ef view caffolding with attributes enabled, all of the attributes are using hard coded strings from the properties names of the related entity.  This is very brittle and can make changes in the entity model extremely difficult and error prone.  The request is to change the code generation logic to output attributes that leverage nameof instead to address this issue.

### Further technical details
EF Core version: 2.1.0-preview2-final
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: Visual Studio 2017 15.4

### PR DESCRIPTION
**Pull request by Muppets**
Summary of the changes
- Switched up ForeignKeyAttribute and InversePropertyAttribute to use nameof() syntax
- Added InversePropertyAttribute into code generation unit test

Fixes #11744

### CODE REVIEW

File: src/EFCore.Design/Scaffolding/Internal/CSharpEntityTypeGenerator.cs [hunk 1]
```diff
@@ -331,9 +331,16 @@ private void GenerateForeignKeyAttribute(INavigation navigation)
                 {
                     var foreignKeyAttribute = new AttributeWriter(nameof(ForeignKeyAttribute));
 
-                    foreignKeyAttribute.AddParameter(
-                        _code.Literal(
-                            string.Join(",", navigation.ForeignKey.Properties.Select(p => p.Name))));
+                    if (navigation.ForeignKey.Properties.Count > 1)
+                    {
+                        foreignKeyAttribute.AddParameter(
```
Comment (smitpatel) [inline src/EFCore.Design/Scaffolding/Internal/CSharpEntityTypeGenerator.cs]: Should we try interpolated strings? (not sure if following works)
`[ForeignKey($"{nameof(Id1)},{nameof(Id2)}")]`

Comment (Muppets) [reply src/EFCore.Design/Scaffolding/Internal/CSharpEntityTypeGenerator.cs]: Ha totally the first thing I tried when writing the code but unforunately attribute arguments must be constants.

image

**Comment by bricelam** [comment]
Blocked on investigating the maximum version of C# we can use (#14545)

**Comment by Muppets** [comment]
As reminded here: https://github.com/aspnet/EntityFrameworkCore/issues/11744#issuecomment-490997376, worth merging this now?

**Pull request merged**